internal/providers/gitlab/webhooksecret/whsecret.go

```diff
@@ -0,0 +1,57 @@
+//
+// Copyright 2024 Stacklok, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package webhooksecret provides a way to generate and verify secrets for GitLab webhooks.
+package webhooksecret
+
+import (
+	sum "crypto/sha512"
```

Comment:
I was wondering a bit if we should use HMAC and treat the secret generation as generating an HMAC for the uniq, but...I can't really convince myself that it makes a difference in this scenario.